packages/ts/src/components/treemap/index.ts

```diff
@@ -0,0 +1,304 @@
+      .data(visibleNodes, d => `${d.data.key}-${d.depth}`)
+
+    const tilesEnter = tiles
+      .enter()
+      .append('g')
+      .attr('class', s.tileGroup)
+
+    // Computes the rect border radius for a given tile.
+    // The rx and ry values are the minimum of the tile
+    // border radius and some fraction the width of the tile,
+    // based on the tileBorderRadiusFactor config.
+    // This ensures that the tile border radius is not
+    // larger than the tile size, which makes small tiles
+    // look better.
+    const rx = (d: TreemapNode<Datum>): number =>
+      Math.min(config.tileBorderRadius, (d.x1 - d.x0) * config.tileBorderRadiusFactor)
+
+    // Tile rectangles
+    tilesEnter
+      .append('rect')
+      .classed(s.tile, true)
+
+      // Make the leaf tiles clickable if a click handler is provided
+      .classed(s.clickableTile, d => config.showTileClickAffordance && !d.children)
+
+      .attr('rx', rx)
+      .attr('ry', rx)
+      // Initialize tile positions so that the initial transition is smooth
+      .attr('x', d => d.x0)
+      .attr('y', d => d.y0)
+      .attr('width', d => d.x1 - d.x0)
+      .attr('height', d => d.y1 - d.y0)
+      .style('fill', d => d._fill ?? getColor(d, config.tileColor))
+      .style('opacity', 0)
+      .style('cursor', config.showTileClickAffordance ? d => !d.children ? 'pointer' : null : null)
+
+
+    const mergedTiles = tiles.merge(tilesEnter)
+
+    smartTransition(mergedTiles.select(`rect.${s.tile}`), duration)
+      .style('fill', d => d._fill ?? getColor(d, config.tileColor))
+      .style('opacity', 1)
+      .attr('x', d => d.x0)
+      .attr('y', d => d.y0)
+      .attr('width', d => d.x1 - d.x0)
+      .attr('height', d => d.y1 - d.y0)
+
+    // Update clipPath rects
+    let svg: Element | null = this.g.node()
+    while (svg && !(svg instanceof SVGSVGElement)) svg = svg.parentElement
+    const defs = svg ? (select(svg).select('defs').empty() ? select(svg).append('defs') : select(svg).select('defs')) : null
+    if (!defs) return
+    const defsSelection = (defs as Selection<SVGDefsElement, unknown, null, undefined>)
+    const clipPaths = defsSelection.selectAll<SVGClipPathElement, TreemapNode<Datum>>('clipPath')
+      .data(visibleNodes, (d: TreemapNode<Datum>) => d._id)
+
+    clipPaths.enter()
+      .append('clipPath')
+      .attr('id', (d: TreemapNode<Datum>) => `clip-${d._id}`)
+      .append('rect')
+      .attr('x', (d: TreemapNode<Datum>) => d.x0)
+      .attr('y', (d: TreemapNode<Datum>) => d.y0)
+      .attr('width', (d: TreemapNode<Datum>) => Math.max(0.1, d.x1 - d.x0))
+      .attr('height', (d: TreemapNode<Datum>) => Math.max(0.1, d.y1 - d.y0))
+      .attr('rx', rx)
+      .attr('ry', rx)
+
+    clipPaths.exit().remove()
+
+    tilesEnter
+      .append('g')
+      .attr('class', 'label-group')
+      .append('text')
+      .attr('class', s.label)
+      .style('opacity', 0)
+
+    mergedTiles.select('g.label-group text')
```

Comment:
I believe that can simply use the `s.label` selector here set as a class name above